src/core/Akka.Persistence.TestKit/JournalWriteBehavior.cs

```diff
@@ -0,0 +1,175 @@
+//-----------------------------------------------------------------------
+// <copyright file="JournalWriteBehavior.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2019 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2019 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Akka.Persistence.TestKit
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public class JournalWriteBehavior : JournalRecoveryBehavior
+    {
+        internal JournalWriteBehavior(IJournalBehaviorSetter setter) : base(setter) { }
+
+        public void Reject() => SetInterceptor(JournalInterceptors.Rejection.Instance);
+
+        public void RejectOnType<TMessage>() => FailOnType(typeof(TMessage));
+
+        public void RejectOnType(Type messageType)
+        {
+            if (messageType is null)
+            {
+                throw new ArgumentNullException(nameof(messageType));
+            }
+
+            SetInterceptor(new JournalInterceptors.OnType(messageType, JournalInterceptors.Rejection.Instance));
+        }
+
+        public void RejectIf(Func<IPersistentRepresentation, bool> predicate)
+        {
+            if (predicate is null)
+            {
+                throw new ArgumentNullException(nameof(predicate));
+            }
+
+            SetInterceptor(new JournalInterceptors.OnCondition(predicate, JournalInterceptors.Rejection.Instance));
```

Comment:
Let's have `SetInterceptor` return a `Task` and have all of these methods `await` on it

Reply:
This make sense...